web/src/copy_messages.ts

```diff
@@ -245,7 +245,22 @@ export function copy_handler(): boolean {
 
     if (!skip_same_td_check && start_id === end_id) {
         // Check whether the selection both starts and ends in the
-        // same message.  If so, Let the browser handle this.
+        // same message. If it does, use the current selection
+        // without needing to consider the complex copy div
+        // construction logic that follows.
+        const div = document.createElement("div");
+        if (selection.rangeCount !== 0) {
+            div.append(selection.getRangeAt(0).cloneContents());
+            const html_content = div.innerHTML.trim();
+            const plain_text = selection.toString().trim();
+            ev.clipboardData?.setData("text/html", html_content);
+            ev.clipboardData?.setData("text/plain", plain_text);
+            div.remove();
+            return true;
+        }
+        // Ideally, this will never happen.
+        // If it does, we let the browser handle it.
+        div.remove();
```

Comment:
Why do we need any of this? Simply `return false` should be enough.

Reply:
I don't see why that would be required -- if we have adjusted what is selected, won't browser's native copying behavior now copy that?

Reply:
I feel like there's something we don't understand about how the browser built-in copy behavior is working.